crates/electrum/src/electrum_ext.rs

```diff
@@ -373,10 +264,16 @@ fn construct_update_tip(
 ///
 /// [tx status](https://electrumx-spesmilo.readthedocs.io/en/latest/protocol-basics.html#status)
 fn determine_tx_anchor(
+    client: &impl ElectrumApi,
     cps: &BTreeMap<u32, CheckPoint>,
     raw_height: i32,
     txid: Txid,
-) -> Option<ConfirmationHeightAnchor> {
+) -> Option<ConfirmationTimeHeightAnchor> {
+    let confirmation_time = client
+        .block_header(raw_height as usize)
+        .expect("header must exist")
```

Comment:
🔧 I think we should handle the error the way we do for the other `ElectrumApi` calls, but since `determine_tx_anchor` doesn't return a `Result`, we could try doing the api call outside the function and pass in the header time on its own, or maybe as a (height, time) tuple.

Reply:
I've been kind of playing around with using a batch command but it's kind of a mess right now, but I'll push a simple fix with your suggestion as it is more clean.